libs/sdk-ui-ext/src/internal/utils/uiConfigHelpers/waterfallChartUiConfigHelper.ts

```diff
@@ -0,0 +1,88 @@
+// (C) 2023 GoodData Corporation
+import cloneDeep from "lodash/cloneDeep";
+import set from "lodash/set";
+import forEach from "lodash/forEach";
+import { IntlShape } from "react-intl";
+
+import { BucketNames } from "@gooddata/sdk-ui";
+import { IExtendedReferencePoint } from "../../interfaces/Visualization";
+
+import { UICONFIG, SUPPORTED, OPEN_AS_REPORT } from "../../constants/uiConfig";
+import { BUCKETS } from "../../constants/bucket";
+
+import { hasMoreThanOneCategory, hasMoreThanOneMasterMeasure } from "./../bucketRules";
+
+import { setBucketTitles } from "./../bucketHelper";
+import { getTranslation } from "./../translations";
+import { hasColorMapping } from "../propertiesHelper";
+import { messages } from "../../../locales";
+
+// If you need to edit these icons
+// reflect changes also in gdc-analytical-designer
+// https://github.com/gooddata/gdc-analytical-designer/blob/develop/app/components/buckets/BucketIcon.tsx
+const waterfallMeasuresIcon = "local:waterfall/bucket-title-measures.svg";
+const waterfallViewIcon = "local:waterfall/bucket-title-view.svg";
+
+function setWaterfallChartBucketWarningMessages(referencePoint: IExtendedReferencePoint, intl?: IntlShape) {
+    const buckets = referencePoint?.buckets;
+    const updatedUiConfig = cloneDeep(referencePoint?.uiConfig);
+
+    forEach(buckets, (bucket) => {
+        const localIdentifier = bucket?.localIdentifier ?? "";
+        const bucketUiConfig = updatedUiConfig?.buckets?.[localIdentifier];
+
+        // skip disabled buckets
+        if (!bucketUiConfig?.enabled) {
+            return;
+        }
+
+        if (!bucketUiConfig?.canAddItems) {
```

Comment:
Please check if this can be simplified to a 
```
if (!bucketUiConfig?.canAddItems && bucket.localIdentifier === BucketNames.VIEW) {
    const warningMessage = getTranslation(messages.category.id, intl);
    set(updatedUiConfig, [BUCKETS, localIdentifier, "warningMessage"], warningMessage);
}
```

Reply:
I updated this. Thank you!